#! /usr/bin/env python
# Task Long configuration

PACKAGE='floor_nav'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description                     Default     Min   Max
gen.add("wait_duration",        double_t, 0,    "Duration in seconds",          1.)
gen.add("new_heading",          double_t, 0,    "Relative angle of heading",    0.)
gen.add("k_theta",              double_t, 0,    "Gain for angular control of the robot angle",  0.05)
gen.add("min_angular_velocity", double_t, 0,    "Min allowed velocity",  		0.05)
gen.add("max_angular_velocity", double_t, 0,    "Max allowed velocity",  		0.1)
gen.add("angle_threshold",      double_t, 0,    "Angle at which the target is considered reached",  0.5)
gen.add("x_milieu",   		    double_t, 0,    "X at which the target is considered reached",  128.0)
gen.add("coeff_pixel",   	  	double_t, 0,    "Reduce Pixel Size",  50.0)


exit(gen.generate(PACKAGE, "task_manager_floor_nav", "TaskStareAtFace"))

